gql/transport/requests.py

```diff
@@ -1,26 +1,66 @@
+from graphql.language.ast import Node
 from graphql.language.printer import print_ast
+from requests.auth import AuthBase
+from requests.cookies import RequestsCookieJar
 
-from .http import HTTPTransport
+from gql.transport.transport import Transport
 
 
-class RequestsHTTPTransport(HTTPTransport):
-    def __init__(self, url, auth=None, use_json=False, timeout=None, **kwargs):
-        """
-        :param url: The GraphQL URL
-        :param auth: Auth tuple or callable to enable Basic/Digest/Custom HTTP Auth
-        :param use_json: Send request body as JSON instead of form-urlencoded
-        :param timeout: Specifies a default timeout for requests (Default: None)
+class RequestsHTTPTransport(Transport):
+    def __init__(
+        self,  # type: RequestsHTTPTransport
+        url,  # type: str
+        headers=None,  # type: Dict[str, Any]
+        cookies=None,  # type: Union[Dict[str, Any], RequestsCookieJar]
+        auth=None,  # type: AuthBase
+        use_json=False,  # type: bool
+        timeout=None,  # type: int
+        verify=True,  # type: bool
+        **kwargs  # type: Any
+    ):
+        """Create a HTTPTransport using requests library to execute GraphQL queries on remote servers.
+
+        :param url: The GraphQL server URL.
+        :param headers: Dictionary of HTTP Headers to send with the :class:`Request` (Default: None).
+        :param cookies: Dict or CookieJar object to send with the :class:`Request` (Default: None).
+        :param auth: Auth tuple or callable to enable Basic/Digest/Custom HTTP Auth (Default: None).
+        :param use_json: Send request body as JSON instead of form-urlencoded (Default: False).
+        :param timeout: Specifies a default timeout for requests (Default: None).
+        :param verify: Either a boolean, in which case it controls whether we verify
+            the server's TLS certificate, or a string, in which case it must be a path
+            to a CA bundle to use. (Default: True).
+        :param kwargs: Optional arguments that ``request`` takes. These can be seen at the :requests_: source code
+            or the official :docs_:
+
+        .. _requests: https://github.com/psf/requests/blob/master/requests/api.py
+        .. _docs: https://requests.readthedocs.io/en/master/
         """
-        super(RequestsHTTPTransport, self).__init__(url, **kwargs)
+        self.url = url
+        self.headers = headers
+        self.cookies = cookies
         self.auth = auth
-        self.default_timeout = timeout
         self.use_json = use_json
+        self.default_timeout = timeout
+        self.verify = verify
+        self.kwargs = kwargs
 
     def execute(self, document, variable_values=None, timeout=None):
+        # type: (Node, dict, int) -> ExecutionResult
+        """Execute the provided document AST for the provided remote server.
+        This make use of requests library within the POST method to perform http calls to the remote server.
```

Comment:
A bit simpler: "This uses the requests library to perform a HTTP POST request to the remote server."